index.js

```diff
@@ -514,6 +514,10 @@ exports.connect = function(config, intern, callback) {
 
   mongoString += host + '/' + config.database;
 
+  if(config.authdb !== undefined) {
+    mongoString += '/?authSource=' + config.authdb;
```

Comment:
When the uriString option is named authSource, then we should also name the options object configuration option like the same thing.